manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java

```diff
@@ -124,6 +124,11 @@ public void deleteBridge(String id, String customerId) {
         }
 
         Bridge bridge = findByIdAndCustomerId(id, customerId);
+        if (!isBridgeDeletable(bridge)) {
+            throw new BridgeLifecycleException("Bridge could only be deleted if its in READY/FAILED state.");
+        }
+        bridge.setStatus(ManagedResourceStatus.DEPROVISION);
```

Comment:
This is done here [too](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/c6eff16620815d47f74ada6483ad18d14256ca20/manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java#L134). Which do you want to keep?

Reply:
@vaibhavjainwiz We don't need this:

```
bridgeDAO.getEntityManager().merge(bridge)
```

The Bridge is already loaded into the `PersistenceContext`, so any changes will be automatically flushed to the DB when the transaction commits. Calling `bridge.setStatus()` is the correct thing to do.

Reply:
Yes, this is probably residue from when I merged the `delete(..)` and `doDelete(..)` methods :+1:

Reply:
@robpblake I am completely agree with you. I was also suprised to see this new changes but keep them to myself to think these changes are made for a reason. 
I will revert them

Reply:
done